Formula/rustfmt.rb

```diff
@@ -0,0 +1,26 @@
+class Rustfmt < Formula
+  desc "Format Rust code"
+  homepage "https://rust-lang.github.io/rustfmt/"
+  url "https://github.com/rust-lang/rustfmt/archive/refs/tags/v1.4.38.tar.gz"
+  sha256 "4856cac540a6d894bc019285b690cebd2076aa8067bd8509ec176645c47a31bf"
+  license any_of: ["MIT", "Apache-2.0"]
+  head "https://github.com/rust-lang/rustfmt.git", branch: "master"
+
+  depends_on "rust" => [:build, :test]
+  depends_on "rustup-init" => :build
```

Comment:
Do we need `rustup` here? Does using `cargo` from the `rust` formula not work?

Reply:
we need `rustup` in order to use nightly toolchain which is required to build this formula from source

Reply:
Is there no way to build it with a stable toolchain?

Reply:
It is documented so in the project readme and I can reproduce the error while building locally without nightly toolchain 

Any suggestion is welcome

Reply:
[link to the readme][1]

[1]: https://github.com/rust-lang/rustfmt/tree/v1.4.38#installing-from-source

Reply:
local reproduction:

```
rustfmt on  HEAD (efa8f55) is 📦 v1.4.38 via 🦀 nightly-2021-10-20-x86_64-apple-darwin
❯ cargo b
   Compiling rustfmt-nightly v1.4.38 (/Users/jiayu_liu/GitHub/rustfmt)
error[E0463]: can't find crate for `rustc_ast`
  --> src/lib.rs:16:1
   |
16 | extern crate rustc_ast;
   | ^^^^^^^^^^^^^^^^^^^^^^^ can't find crate

For more information about this error, try `rustc --explain E0463`.
error: could not compile `rustfmt-nightly` due to previous error
```

`rustc_ast` is nightly only